packages/core/src/modules/non-secret/service/NonSecretRecordService.ts

```diff
@@ -0,0 +1,40 @@
+import type { ConnectionRecord } from '../../connections/repository/ConnectionRecord'
+import type { NonSecretRecordTags } from '../repository/NonSecretRecord'
+
+import { Lifecycle, scoped } from 'tsyringe'
+
+import { AriesFrameworkError } from '../../../error'
+import { NonSecretRecord } from '../repository/NonSecretRecord'
+import { NonSecretRecordRepository } from '../repository/NonSecretRecordRepository'
+
+@scoped(Lifecycle.ContainerScoped)
+export class NonSecretRecordService {
+  private nonSecretRepository: NonSecretRecordRepository
+
+  public constructor(nonSecretRepository: NonSecretRecordRepository) {
+    this.nonSecretRepository = nonSecretRepository
+  }
+
+  public async saveRecord(message: string, tags?: NonSecretRecordTags, connectionRecord?: ConnectionRecord) {
```

Comment:
It would be better for the api to take an object so that we have named parameters.

Somehting like this: 

```suggestion
public async saveRecord({message, tags, connectionRecord}: SaveNonSecretRecordOption) {
```